src/layer.rs

```diff
@@ -195,12 +213,21 @@ impl LayerSet {
             // Dance around the borrow checker by using indices instead of references.
             let layer_pos = self.layers.iter().position(|l| l.name.as_ref() == old).unwrap();
 
-            // Default layer: just change the name. Non-default layer: change name and path.
-            let layer = &mut self.layers[layer_pos];
-            if layer_pos != 0 {
-                layer.path = crate::util::default_file_name_for_layer_name(&name).into();
+            if layer_pos == 0 {
+                // Default layer: just change the name.
+                let layer = &mut self.layers[layer_pos];
+                layer.name = name;
+            } else {
+                // Non-default layer.
+                let old_path = self.layers[layer_pos].path.to_string_lossy().to_lowercase();
+                self.path_set.remove(&old_path);
+                let new_path = crate::util::default_file_name_for_layer_name(&name, &self.path_set);
+                self.path_set.insert(new_path.to_string_lossy().to_lowercase());
+
+                let layer = &mut self.layers[layer_pos];
+                layer.name = name;
+                layer.path = new_path;
```

Comment:
ditto, this whole block can just be:

```rust
let layer_pos = self.layers.iter().position(|l| l.name.as_ref() == old).unwrap();

// change path only if not the default layer:
if layer_pos != 0 {
    let old_path = self.layers[layer_pos].path.to_string_lossy().to_lowercase();
    self.path_set.remove(&old_path);
    let new_path = crate::util::default_file_name_for_layer_name(&name, &self.path_set);
    self.path_set.insert(new_path.to_string_lossy().to_lowercase());
    self.layers[layer_pos].path = new_path;
}

self.layers[layer_pos].name = name;
```

do you understand the rationale here? it's about having the only code in the conditional block be the code that is 'special' to the condition, which makes the code easier to read overall; this is a good practice to follow whenever possible.

Reply:
Ah duh, I didn't see that. Thanks!